apps/test/unit/lib/levelbuilder/code-docs-editor/ProgrammingMethodEditorTest.jsx

```diff
@@ -0,0 +1,177 @@
+    expect(
+      wrapper
+        .find('input')
+        .at(1)
+        .props().value
+    ).to.equal('painter');
+    expect(
+      wrapper
+        .find('input')
+        .at(1)
+        .props().readOnly
+    ).to.be.true;
+  });
+
+  it('displays initial values in input fields in documentation section', () => {
+    const wrapper = shallow(
+      <ProgrammingMethodEditor
+        initialProgrammingMethod={initialProgrammingMethod}
+      />
+    );
+    expect(wrapper.text().includes('Editing Method')).to.be.true;
+
+    // Documentation section
+    const documentationSection = wrapper.find('CollapsibleEditorSection').at(0);
+    expect(documentationSection.props().title).to.equal('Documentation');
+    expect(
+      documentationSection
+        .find('input')
+        .at(0)
+        .props().value
+    ).to.equal('developer.mozilla.org');
+    expect(
+      documentationSection
+        .find('TextareaWithMarkdownPreview')
+        .at(0)
+        .props().markdown
+    ).to.equal('This is a longer description of the code.');
+  });
+
+  it('displays initial values in details section', () => {
+    const wrapper = shallow(
+      <ProgrammingMethodEditor
+        initialProgrammingMethod={initialProgrammingMethod}
+      />
+    );
+    const detailsSection = wrapper.find('CollapsibleEditorSection').at(1);
+    expect(
+      detailsSection
+        .find('TextareaWithMarkdownPreview')
+        .at(0)
+        .props().markdown
+    ).to.equal('Painter()');
+  });
+
+  it('displays initial values in input fields in parameters section', () => {
+    const wrapper = shallow(
+      <ProgrammingMethodEditor
+        initialProgrammingMethod={initialProgrammingMethod}
+      />
+    );
+    expect(wrapper.text().includes('Editing Method')).to.be.true;
+    // Parameters section
+    const parametersSection = wrapper.find('CollapsibleEditorSection').at(2);
+    expect(parametersSection.props().title).to.equal('Parameters');
+    const orderableParameterList = parametersSection.find('OrderableList');
+    expect(orderableParameterList.props().addButtonText).to.equal(
+      'Add Another Parameter'
+    );
+    expect(orderableParameterList.props().list.length).to.equal(1);
+  });
+
+  it('displays initial values in input fields in examples section', () => {
+    const wrapper = shallow(
+      <ProgrammingMethodEditor
+        initialProgrammingMethod={initialProgrammingMethod}
+      />
+    );
+    expect(wrapper.text().includes('Editing Method')).to.be.true;
+    // Examples section
+    const examplesSection = wrapper.find('CollapsibleEditorSection').at(3);
+    expect(examplesSection.props().title).to.equal('Examples');
+    const orderableExampleList = examplesSection.find('OrderableList');
+    expect(orderableExampleList.props().addButtonText).to.equal(
+      'Add Another Example'
+    );
+    expect(orderableExampleList.props().list.length).to.equal(1);
+  });
+
+  it('attempts to save when save is pressed', () => {
```

Comment:
The reason I didn't test editing specifically is that we don't need to test that an input field is calling an `onChange` handler (we can just assume it will). However, there is some logic as to what is in that `onChange` handler that we could test ... except I'm not sure the best way to do that.

For example, I might want to change that the `name` input field updates `name` when changed (and not something else). But I'm having trouble figuring out how to test that for components, such as `TextareaWithMarkdownPreview`.

I'm going to update this test to edit a couple of fields -- if nothing else, it'll test that we don't just try to save the initial state again. I'm not super comfortable with React testing in general so let me know if there's something else I should be testing!

Reply:
I'm not super comfy either :) -- I think verifying some sort of (minimal) edit behavior sounds like a reasonable approach.